File: HW/HW11/HW11.py
class Solution:
    def floodFill(self, image, sr, sc, newColor):
        if not image or image[sr][sc] == newColor:
            return image
        
        oldColor = image[sr][sc]
        self.dfs(image, sr, sc, oldColor, newColor)
        return image
    
    def dfs(self, image, r, c, oldColor, newColor):
        if r < 0 or r >= len(image) or c < 0 or c >= len(image[0]) or image[r][c] != oldColor:
            return
        image[r][c] = newColor #
        self.dfs(image, r + 1, c, oldColor, newColor)
        self.dfs(image, r - 1, c, oldColor, newColor)
        self.dfs(image, r, c + 1, oldColor, newColor)
        self.dfs(image, r, c - 1, oldColor, newColor)

File: HW/HW11/HW11.java
public class HW11 {
    public int[][] floodFill(int[][] image, int sr, int sc, int newColor) {
        if (image[sr][sc] == newColor) {
            return image;
        }
        int oldColor = image[sr][sc];
        dfs(image, sr, sc, oldColor, newColor);
        return image;
    }
    private void dfs(int[][] image, int r, int c, int oldColor, int newColor) {
        if (r < 0 || r >= image.length || c < 0 || c >= image[0].length || image[r][c] != oldColor) {
            return;
        }
        image[r][c] = newColor;
        dfs(image, r + 1, c, oldColor, newColor);
        dfs(image, r - 1, c, oldColor, newColor);
        dfs(image, r, c + 1, oldColor, newColor);
        dfs(image, r, c - 1, oldColor, newColor);
    }
}

